src/engine.h

```diff
@@ -156,6 +157,11 @@ class Engine {
   /// commands have occurred e.g., DoClear, DoDrawArrays, DoDrawRect.
   virtual Result GetFrameBufferInfo(ResourceInfo* info) = 0;
 
+  /// Extract the pixel values from the framebuffer into the |pixels|
+  /// DataBuffer storing pixels value as a row-major array of RGBA
+  /// channels. The DatumType indicates the format of each channel.
+  virtual Result GetFrameBuffer(FormatBuffer* pixels) = 0;
```

Comment:
`class EngineStub : public Engine` in src/executor_test.cc needs to override this method.
Please add `Result GetFrameBuffer(FormatBuffer*) override { return {}; }` to `class EngineStub : public Engine`.